Action freezes on ubuntu-20.04 image

The build on ubuntu-20.04 image just freezes.
Step code:
```
  build-linux-snap:
    runs-on: ubuntu-20.04
    steps:
      - uses: actions/checkout@v2
        with:
          submodules: recursive
      - uses: snapcore/action-build@v1
```

Step log:
```
2020-06-10T10:49:57.8182042Z ##[group]Run snapcore/action-build@v1
2020-06-10T10:49:57.8182175Z with:
2020-06-10T10:49:57.8182445Z   path: .
2020-06-10T10:49:57.8182533Z   build-info: true
2020-06-10T10:49:57.8182622Z ##[endgroup]
2020-06-10T10:49:57.8498536Z Building Snapcraft project in "."...
2020-06-10T10:49:57.8505981Z ##[group]Installing Snapcraft plus dependencies
2020-06-10T10:49:57.8547415Z Installing Snapcraft...
2020-06-10T10:49:57.8578660Z [command]/usr/bin/sudo snap install --classic snapcraft
2020-06-10T10:50:06.1783719Z snapcraft 4.0.4 from Canonical* installed
2020-06-10T10:50:06.1816518Z ##[endgroup]
2020-06-10T10:50:06.1831886Z [command]/usr/bin/sg lxd -c snapcraft
2020-06-10T16:49:47.5563821Z Password: 
2020-06-10T16:49:47.5777857Z ##[error]The operation was canceled.
2020-06-10T16:49:47.5831445Z Post job cleanup.
```

Yes, the build took 6 hours, and just timed out. There were no logs at all during the build (the step log was not expandable).
Apparently, starting LXD container requires password now (or something like that), even on passwordless sudo.
Changing the runner to `ubuntu-18.04` "fixed" the problem, but, of course, the support for Ubuntu 20.04 should be added.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -14,6 +14,7 @@ jobs:
     - run: |
         npm install
         npm run all
+
   integration: # make sure the action works on a clean machine without building
     runs-on: ubuntu-latest
     strategy:
@@ -32,3 +33,17 @@ jobs:
       with:
         name: 'snap'
         path: ${{ steps.snapcraft.outputs.snap}}
+
+  check-runner: # make sure the action works on each VM image
+    strategy:
+      matrix:
+        runner:
+          - ubuntu-16.04
+          - ubuntu-18.04
+          - ubuntu-20.04
+    runs-on: ${{ matrix.runner }}
+    steps:
+    - uses: actions/checkout@v2
+    - uses: ./
+      with:
+        path: './test-projects/core18'

--- a/__tests__/tools.test.ts
+++ b/__tests__/tools.test.ts
@@ -177,8 +177,8 @@ test('ensureLXD removes the apt version of LXD', async () => {
   ])
 })
 
-test('ensureLXD is a no-op if LXD is installed', async () => {
-  expect.assertions(3)
+test('ensureLXD still calls "lxd init" if LXD is installed', async () => {
+  expect.assertions(4)
 
   const accessMock = jest.spyOn(fs.promises, 'access').mockImplementation(
     async (filename: fs.PathLike, mode?: number | undefined): Promise<void> => {
@@ -210,6 +210,7 @@ test('ensureLXD is a no-op if LXD is installed', async () => {
     'lxd',
     os.userInfo().username
   ])
+  expect(execMock).toHaveBeenNthCalledWith(3, 'sudo', ['lxd', 'init', '--auto'])
 })
 
 test('ensureSnapcraft installs Snapcraft if needed', async () => {

--- a/__tests__/tools.test.ts
+++ b/__tests__/tools.test.ts
@@ -114,7 +114,7 @@ test('ensureSnapd fixes permissions on the root directory', async () => {
 })
 
 test('ensureLXD installs the snap version of LXD if needed', async () => {
-  expect.assertions(4)
+  expect.assertions(5)
 
   const accessMock = jest.spyOn(fs.promises, 'access').mockImplementation(
     async (filename: fs.PathLike, mode?: number | undefined): Promise<void> => {
@@ -129,20 +129,26 @@ test('ensureLXD installs the snap version of LXD if needed', async () => {
 
   await tools.ensureLXD()
 
-  expect(accessMock).toHaveBeenCalled()
   expect(execMock).toHaveBeenNthCalledWith(1, 'sudo', [
-    'snap',
-    'install',
+    'groupadd',
+    '--force',
+    '--system',
     'lxd'
   ])
-  expect(execMock).toHaveBeenNthCalledWith(2, 'sudo', ['lxd', 'init', '--auto'])
-  expect(execMock).toHaveBeenNthCalledWith(3, 'sudo', [
+  expect(execMock).toHaveBeenNthCalledWith(2, 'sudo', [
     'usermod',
     '--append',
     '--groups',
     'lxd',
     os.userInfo().username
   ])
+  expect(accessMock).toHaveBeenCalled()
+  expect(execMock).toHaveBeenNthCalledWith(3, 'sudo', [
+    'snap',
+    'install',
+    'lxd'
+  ])
+  expect(execMock).toHaveBeenNthCalledWith(4, 'sudo', ['lxd', 'init', '--auto'])
 })
 
 test('ensureLXD removes the apt version of LXD', async () => {
@@ -172,7 +178,7 @@ test('ensureLXD removes the apt version of LXD', async () => {
 })
 
 test('ensureLXD is a no-op if LXD is installed', async () => {
-  expect.assertions(2)
+  expect.assertions(3)
 
   const accessMock = jest.spyOn(fs.promises, 'access').mockImplementation(
     async (filename: fs.PathLike, mode?: number | undefined): Promise<void> => {
@@ -191,7 +197,19 @@ test('ensureLXD is a no-op if LXD is installed', async () => {
   await tools.ensureLXD()
 
   expect(accessMock).toHaveBeenCalled()
-  expect(execMock).not.toHaveBeenCalled()
+  expect(execMock).toHaveBeenNthCalledWith(1, 'sudo', [
+    'groupadd',
+    '--force',
+    '--system',
+    'lxd'
+  ])
+  expect(execMock).toHaveBeenNthCalledWith(2, 'sudo', [
+    'usermod',
+    '--append',
+    '--groups',
+    'lxd',
+    os.userInfo().username
+  ])
 })
 
 test('ensureSnapcraft installs Snapcraft if needed', async () => {

--- a/src/tools.ts
+++ b/src/tools.ts
@@ -35,18 +35,23 @@ export async function ensureLXD(): Promise<void> {
     core.info('Removing legacy .deb packaged LXD...')
     await exec.exec('sudo', ['apt-get', 'remove', '-qy', 'lxd', 'lxd-client'])
   }
+
+  core.info(`Ensuring ${os.userInfo().username} is in the lxd group...`)
+  await exec.exec('sudo', ['groupadd', '--force', '--system', 'lxd'])
+  await exec.exec('sudo', [
+    'usermod',
+    '--append',
+    '--groups',
+    'lxd',
+    os.userInfo().username
+  ])
+
+  // Ensure that the "lxd" group exists
   const haveSnapLXD = await haveExecutable('/snap/bin/lxd')
   if (!haveSnapLXD) {
     core.info('Installing LXD...')
     await exec.exec('sudo', ['snap', 'install', 'lxd'])
     await exec.exec('sudo', ['lxd', 'init', '--auto'])
-    await exec.exec('sudo', [
-      'usermod',
-      '--append',
-      '--groups',
-      'lxd',
-      os.userInfo().username
-    ])
   }
 }
 

--- a/src/tools.ts
+++ b/src/tools.ts
@@ -51,8 +51,9 @@ export async function ensureLXD(): Promise<void> {
   if (!haveSnapLXD) {
     core.info('Installing LXD...')
     await exec.exec('sudo', ['snap', 'install', 'lxd'])
-    await exec.exec('sudo', ['lxd', 'init', '--auto'])
   }
+  core.info('Initialising LXD...')
+  await exec.exec('sudo', ['lxd', 'init', '--auto'])
 }
 
 export async function ensureSnapcraft(): Promise<void> {

